s3api/server.go

```diff
@@ -180,7 +182,16 @@ func globalErrorHandler(ctx *fiber.Ctx, er error) error {
 		// log it as a panic log
 		debuglogger.Panic(er)
 	} else {
-		// otherwise log it as an internal error
+		// handle the fiber specific errors
+		var fiberErr *fiber.Error
+		if errors.As(er, &fiberErr) {
+			if strings.Contains(fiberErr.Message, "cannot parse Content-Length") {
```

Comment:
This is a bummer that we have to string match for the error here. I dont see any other way to do this though. Is it possible to somehow stick in our own content length check before fasthttp does its header checks? Or maybe we could try to submit some changes to fasthttp to return types errors for some of these cases that we need to specifically handle.